### PR TITLE
DVS-3200 Document DVS workaround for csm-1.5 upgrade

### PR DESCRIPTION
**Pull request by paulcassella-hpe**
# Description

Document the workaround of extracting and copying to worker nodes the DVS prechecks_for_worker_reboots when upgrading from COS older than 2.5.146.

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Comment by paulcassella-hpe** [comment]
force-pushed to address two markdown style errors.

The other style complaints are that the links I added aren't valid, but I don't know what's wrong with them.

I also don't know why the spellchecker is complaining about "csm" and "cd" and so on.

File: operations/iuf/workflows/management_rollout.md [hunk 1]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
+5. (`ncn-m001#`) After the canary node has booted successfully, install the new version of the script onto the other worker nodes. This is one way to do that:
+
+    ```bash
+    for name in $(kubectl get node | grep -P 'ncn-w\d+' | awk '{print $1}'); do
+        xname=$(nslookup $(dig +short ${name}.nmn) | grep -P "x\d+c\d+s\d+b\dn\d.$" | sed -e 's/.* = //' -e 's/\.$//')
+        scp $SSH_OPTIONS opt/cray/shasta/cne/bin/prechecks_for_worker_reboots $xname:/opt/cray/shasta/cos/bin/prechecks_for_worker_reboots
+    done
+    ```
+
+6. (`ncn-m001#`) Optionally remove the temporary directory
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
6. (`ncn-m001#`) Optionally, remove the temporary directory.
```

File: operations/iuf/workflows/management_rollout.md [hunk 2]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
+    cd /tmp/upgrade-prechecks_WAR
+    ```
+
+3. (`ncn-m001#`) Extract the cray-dvs-csm rpm that's included in the uss image:
+
+    ```bash
+    rpm2cpio < $MEDIADIR/uss-*-csm-1.5/rpms/uss-*-csm-1.5/x86_64/cray-dvs-csm-*.x86_64.rpm | cpio -i --make-directories --no-absolute-filenames
+    ```
+
+4. (`ncn-m001#`) Install the new version of the script onto the canary node
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
4. (`ncn-m001#`) Install the new version of the script onto the canary node.
```

File: operations/iuf/workflows/management_rollout.md [hunk 3]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
+
+2. (`ncn-m001#`) Optionally create and cd to a temporary directory
+in which to extract the new version of the script.
+
+    ```bash
+    mkdir /tmp/upgrade-prechecks_WAR
+    cd /tmp/upgrade-prechecks_WAR
+    ```
+
+3. (`ncn-m001#`) Extract the cray-dvs-csm rpm that's included in the uss image:
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
3. (`ncn-m001#`) Extract the `cray-dvs-csm` rpm that's included in the uss image:
```

File: operations/iuf/workflows/management_rollout.md [hunk 4]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
+rebuilt, and to the remaining worker nodes after the canary node boot
+has succeeded.
+
+1. (`ncn-m001#`) Optionally set an environment variable to the media directory:
+
+    ```bash
+    MEDIADIR=/etc/cray/upgrade/csm/media/<directory>
+    ```
+
+2. (`ncn-m001#`) Optionally create and cd to a temporary directory
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
2. (`ncn-m001#`) Optionally, create and `cd` to a temporary directory.
```

File: operations/iuf/workflows/management_rollout.md [hunk 5]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
+The new version of the script may be found in the cray-dvs-csm rpm in
+the uss csm tar file in the upgrade's media directory.  The workaround
+is to extract the script from the rpm to a temporary directory and
+then copy it to the worker nodes.
+
+It should be copied to the canary node when that node is being
+rebuilt, and to the remaining worker nodes after the canary node boot
+has succeeded.
+
+1. (`ncn-m001#`) Optionally set an environment variable to the media directory:
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
1. (`ncn-m001#`) Optionally, set an environment variable to the media directory:
```

File: operations/iuf/workflows/management_rollout.md [hunk 6]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
...
 [Management-nodes-rollout without CSM upgrade](#22-management-nodes-rollout-without-csm-upgrade).
 
+#### 2.3.1 DVS workaround upgrading from COS prior to 2.5.146
+
+If COS prior to 2.5.146 is installed prior to upgrading to CSM-1.5, the management rollout in this step may hang.  There is a workaround for this, copying the new version of the DVS
+prechecks_for_worker_reboots script to all NCN worker nodes as /opt/cray/shasta/cos/bin/prechecks_for_worker_reboots
+
+This is to be run on the ncn-m001 node during this step 2.3 NCN worker nodes.
+
+The new version of the script may be found in the cray-dvs-csm rpm in
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
The new version of the script may be found in the `cray-dvs-csm` rpm in
```

File: operations/iuf/workflows/management_rollout.md [hunk 7]
```diff
@@ -288,6 +292,65 @@ Once this step has completed:
 Return to the procedure that was being followed for `management-nodes-rollout` to complete the next step, either [Management-nodes-rollout with CSM upgrade](#21-management-nodes-rollout-with-csm-upgrade) or
 [Management-nodes-rollout without CSM upgrade](#22-management-nodes-rollout-without-csm-upgrade).
 
+#### 2.3.1 DVS workaround upgrading from COS prior to 2.5.146
+
+If COS prior to 2.5.146 is installed prior to upgrading to CSM-1.5, the management rollout in this step may hang.  There is a workaround for this, copying the new version of the DVS
+prechecks_for_worker_reboots script to all NCN worker nodes as /opt/cray/shasta/cos/bin/prechecks_for_worker_reboots
+
+This is to be run on the ncn-m001 node during this step 2.3 NCN worker nodes.
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
This is to be run on the `ncn-m001` node during this step 2.3 NCN worker nodes.
```

File: operations/iuf/workflows/management_rollout.md [hunk 8]
```diff
@@ -235,6 +236,9 @@ for details on how to query the images and CFS configurations and see the [updat
 **`NOTE`** The `management-nodes-rollout` stage creates additional separate Argo workflows when rebuilding NCN worker nodes. The Argo workflow names will include the string `ncn-lifecycle-rebuild`. If monitoring progress with the Argo UI,
 remember to include these workflows.
 
+**`NOTE`** If upgrading from CSM-1.4 to CSM-1.5 with a COS release prior to 2.5.146 currently installed, a workaround is needed to roll out the management nodes.  See the later subsection [2.3.1 DVS workaround upgrading from COS prior to
+2.5.146](#231-DVS-workaround-upgrading-from-COS-prior-to-25146).  If the installed COS is 2.5.146 or later, this is not needed.
```
Comment (nrockershousen) [inline operations/iuf/workflows/management_rollout.md]:
```suggestion
2.5.146](#231-DVS-workaround-upgrading-from-COS-prior-to-25146).  If the installed COS version is 2.5.146 or later, this is not needed.
```

**Review by rustydb — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Review by pshields2 — APPROVED** [review]

**Review by cliffwickman — APPROVED** [review]
I am not familiar with the 'canary' terminology, but I assume that someone doing and install is familiar. It must be used in the install documentation.
It looks clear and explicit enough to be easily followed.  I approve.

**Comment by paulcassella-hpe** [comment]
This WAR has been verified via use in the upgrade of an internal system.

**Pull request merged**